bittensor/core/async_subtensor.py

```diff
@@ -152,14 +159,106 @@ async def encode_params(
+        """
+        return await self.substrate.get_block_hash(
+            block_id if block_id else await self.get_current_block()
+        )
+
+    async def is_hotkey_registered_any(
+        self, hotkey_ss58: str, block_hash: Optional[str] = None
+    ) -> bool:
+        """
+        Checks if a neuron's hotkey is registered on any subnet within the Bittensor network.
+
+        Args:
+            hotkey_ss58 (str): The ``SS58`` address of the neuron's hotkey.
+            block_hash (Optional[str]): The blockchain block_hash representation of block id.
+
+        Returns:
+            bool: ``True`` if the hotkey is registered on any subnet, False otherwise.
+
+        This function is essential for determining the network-wide presence and participation of a neuron.
+        """
+        return len(await self.get_netuids_for_hotkey(hotkey_ss58, block_hash)) > 0
+
+    async def get_subnet_burn_cost(
         self, block_hash: Optional[str] = None
-    ) -> list[int]:
+    ) -> Optional[str]:
+        """
+        Retrieves the burn cost for registering a new subnet within the Bittensor network. This cost represents the amount of Tao that needs to be locked or burned to establish a new subnet.
+
+        Args:
+            block_hash (Optional[int]): The blockchain block_hash of the block id.
+
+        Returns:
+            int: The burn cost for subnet registration.
+
+        The subnet burn cost is an important economic parameter, reflecting the network's mechanisms for controlling the proliferation of subnets and ensuring their commitment to the network's long-term viability.
+        """
+        lock_cost = await self.query_runtime_api(
+            runtime_api="SubnetRegistrationRuntimeApi",
+            method="get_network_registration_cost",
+            params=[],
+            block_hash=block_hash,
+        )
+
+        return lock_cost
+
+    async def get_total_subnets(
+        self, block_hash: Optional[str] = None
+    ) -> Optional[int]:
+        """
+        Retrieves the total number of subnets within the Bittensor network as of a specific blockchain block.
+
+        Args:
+            block_hash (Optional[str]): The blockchain block_hash representation of block id.
+
+        Returns:
+            Optional[str]: The total number of subnets in the network.
+
+        Understanding the total number of subnets is essential for assessing the network's growth and the extent of its decentralized infrastructure.
+        """
+        result = await self.substrate.query(
+            module="SubtensorModule",
+            storage_function="TotalNetworks",
+            params=[],
+            block_hash=block_hash
+            if block_hash is not None
+            else await self.get_block_hash(await self.get_current_block()),
```

Comment:
```suggestion
            block_hash=block_hash
```

It's unnecessary to retrieve the block hash in this, as it will automatically be done in the query if not specified.